pkg/orchestrator/gitHubActions.go

```diff
@@ -191,26 +221,83 @@ func isGitHubActions() bool {
 	return areIndicatingEnvVarsSet(envVars)
 }
 
-func (g *GitHubActionsConfigProvider) getStageIds() ([]int, error) {
-	resp, err := g.client.GetRequest(fmt.Sprintf("%s/runs/%s/jobs", getActionsURL(), getEnv("GITHUB_RUN_ID", "")), httpHeader, nil)
+// actionsURL returns URL to actions resource. For example,
+// https://api.github.com/repos/SAP/jenkins-library/actions              - if it's github.com
+// https://github.tools.sap/api/v3/repos/project-piper/sap-piper/actions - if it's GitHub Enterprise
+func actionsURL() string {
+	return getEnv("GITHUB_API_URL", "") + "/repos/" + getEnv("GITHUB_REPOSITORY", "") + "/actions"
+}
+
+func (g *GitHubActionsConfigProvider) fetchRunData() {
+	if g.runData.fetched {
+		return
+	}
+
+	url := fmt.Sprintf("%s/runs/%s", actionsURL(), getEnv("GITHUB_RUN_ID", ""))
+	resp, err := g.client.GetRequest(url, httpHeaders, nil)
+	if err != nil || resp.StatusCode != 200 {
+		log.Entry().Errorf("failed to get API data: %s", err)
+		return
+	}
+
+	err = piperHttp.ParseHTTPResponseBodyJSON(resp, &g.runData)
 	if err != nil {
-		return nil, fmt.Errorf("failed to get API data: %w", err)
+		log.Entry().Errorf("failed to parse JSON data: %s", err)
+		return
+	}
+	g.runData.fetched = true
+}
+
+func (g *GitHubActionsConfigProvider) fetchJobs() error {
+	if g.jobsFetched {
+		return nil
 	}
 
-	var stagesID stagesID
-	err = piperHttp.ParseHTTPResponseBodyJSON(resp, &stagesID)
+	url := fmt.Sprintf("%s/runs/%s/jobs", actionsURL(), getEnv("GITHUB_RUN_ID", ""))
+	resp, err := g.client.GetRequest(url, httpHeaders, nil)
```

Comment:
I would like to see GH api interactions to be implemented in the `github` pkg, in fact we already use the go-github package there which should be able to shield away the actual api calls.